pkg/queue/cluster_queue_impl.go

```diff
@@ -221,6 +221,21 @@ func (c *clusterQueueBase) DumpInadmissible() (sets.Set[string], bool) {
 	return elements, true
 }
 
+func (c *clusterQueueBase) Snapshot(maxCount int32) ([]*kueue.Workload, bool) {
+	if c.heap.Len() == 0 {
+		return nil, false
+	}
+	elements := make([]*kueue.Workload, 0, c.heap.Len())
+	for i, e := range c.heap.List() {
+		if int32(i) > maxCount {
```

Comment:
We need to sort the merge(heap.items, inadmissibleWorkloads) before we can cut off.